app/src/main/java/fr/free/nrw/commons/media/MediaClient.java

```diff
@@ -83,6 +84,24 @@ public MediaClient(MediaInterface mediaInterface, MediaDetailInterface mediaDeta
 
     }
 
+    /**
+     * This method takes the userName as input and returns a list of  Media objects filtered using
+     * allimages query It uses the allimages query API to get the images contributed by the userName,
+     * 10 at a time.
+     *
+     * @param userName the username
+     * @return
+     */
+    public Single<List<Media>> getMediaListForUser(String userName) {
+        return responseToMediaList(
+            continuationStore.containsKey("user_" + userName) ?
```

Comment:
this is a bit hard on the eyes, how about moving the ternary into the method call?

```
mediaInterface.getMediaListForUser(
username,
10,
continuationStore.hasUser ? continuation.getUser : emptyMap()
)
```